src/ser/mod.rs

```diff
@@ -401,19 +437,70 @@ mod tests {
     }
 
     #[test]
-    #[ignore]
-    fn serialize_a_list() {
-        let inputs = vec![1, 2, 3, 4];
+    fn test_serialize_vector() {
+        let vector = vec![1, 2, 3];
+        let mut buffer = Vec::new();
+        {
+            let mut ser = Serializer::new(&mut buffer);
+            vector.serialize(&mut ser).unwrap();
+        }
+        let got = String::from_utf8(buffer).unwrap();
+        assert_eq!("123", got);
+    }
+
+    #[test]
+    fn test_serialize_vector_of_primitives_within_struct() {
+        #[derive(Serialize)]
+        struct Thing {
+            things: Vec<String>,
+        }
+
+        let sut = Thing {
+            things: vec!["thing_1".to_string(), "thing two".to_string()],
+        };
 
         let mut buffer = Vec::new();
+        {
+            let mut ser = Serializer::new(&mut buffer);
+            sut.serialize(&mut ser).unwrap();
+        }
 
+        let got = String::from_utf8(buffer).unwrap();
+        assert_eq!("<Thing><things>thing_1thing two</things></Thing>", got);
```

Comment:
Not sure if this is the expected behavior, but we can't do much to separate the vector items if we don't have any tag. To properly serialize vectors we should use types that wraps a vector such as `MyStruct(Vec<String>)`.